File: JavaScript/removeDupes.js
const removeDuplicates = (arr) => {
	const hash = {};

	for (let i = 0; i < arr.length; i++) {
		let current = arr[i];

		if (!hash[current]) {
			hash[current] = arr[i];
		}
	}

	return Object.values(hash);
};
// test
const array = [1, 1, 2, 2, 2, 3, 3];
console.log(removeDuplicates(array));

File: python/remove_dupes.py
# Solution 1


# def remove_duplicates(arr):
#     val_tracker = {}

#     for num in arr:
#         if num not in arr:
#             val_tracker[num] = 1
#         else:
#             print('True')
#             return True
#     print('False')
#     return False

# Solution 2


def remove_duplicates(arr):
    unique = set(arr)

    if len(unique) != len(arr):
        print('True')
        return True
    else:
        print('False')
        return False


test_arr = [1, 1, 1, 1, 2, 3, 4, 5, 5, 5]
answer_arr = set(test_arr)
remove_duplicates(test_arr)
